test/membrane_rtmp_plugin/rtmp_sink_test.exs

```diff
@@ -97,37 +126,52 @@ defmodule Membrane.RTMP.SinkTest do
     Pipeline.start_link_supervised!(options)
   end
 
-  defp start_sink_pipeline(rtmp_url) do
+  defp start_sink_pipeline(rtmp_url, tracks) do
     import Membrane.ChildrenSpec
 
     options = [
-      structure: [
-        child(:rtmp_sink, %Membrane.RTMP.Sink{rtmp_url: rtmp_url, max_attempts: 5}),
-        #
-        child(:video_source, %Membrane.Hackney.Source{
-          location: @input_video_url,
-          hackney_opts: [follow_redirect: true]
-        })
-        |> child(:video_parser, %Membrane.H264.FFmpeg.Parser{
-          framerate: {30, 1},
-          alignment: :au,
-          attach_nalus?: true,
-          skip_until_parameters?: false
-        })
-        |> child(:video_payloader, Membrane.MP4.Payloader.H264)
-        |> via_in(:video)
-        |> get_child(:rtmp_sink),
-        #
-        child(:audio_source, %Membrane.Hackney.Source{
-          location: @input_audio_url,
-          hackney_opts: [follow_redirect: true]
-        })
-        |> child(:audio_parser, %Membrane.AAC.Parser{
-          out_encapsulation: :none
-        })
-        |> via_in(:audio)
-        |> get_child(:rtmp_sink)
-      ],
+      structure:
+        [
+          child(:rtmp_sink, %Membrane.RTMP.Sink{
+            rtmp_url: rtmp_url,
+            tracks: tracks,
+            max_attempts: 5
+          })
+        ] ++
+          if :audio in tracks do
```

Comment:
nitpick: just for readability we can probably extract this (+ the video below) to a helper function with a guard `when :video in tracks/ when :audio in tracks`

Reply:
Unfortunately, I cannot use guards, because the `tracks` are unknown at compile time. I will move it to a separate function though.